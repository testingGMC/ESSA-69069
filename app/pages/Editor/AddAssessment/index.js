import AddAssessment from './AddAssessment'

export default AddAssessment
