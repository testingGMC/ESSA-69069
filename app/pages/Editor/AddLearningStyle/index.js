import AddLearningStyle from './AddLearningStyle'

export default AddLearningStyle
