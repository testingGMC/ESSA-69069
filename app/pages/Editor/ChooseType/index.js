import ChooseType from './ChooseType'

export default ChooseType
