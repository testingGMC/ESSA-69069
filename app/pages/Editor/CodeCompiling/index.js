import CodeCompiling from './CodeCompiling'

export default CodeCompiling
