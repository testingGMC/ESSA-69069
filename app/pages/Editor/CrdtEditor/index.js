import CrdtEditor from './CrdtEditor'

export default CrdtEditor
