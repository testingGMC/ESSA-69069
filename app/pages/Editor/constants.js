export const ASSESSMENTS = {
  0: { key: 0, title: 'Quiz', formattedMessage: 'quiz', icon: 'snippets' },
  1: {
    key: 1,
    title: 'Reorder challenge',
    formattedMessage: 'reorderChallenge',
    icon: 'database',
  },
  2: {
    key: 2,
    title: 'Fill in the blank',
    formattedMessage: 'fillInTheBlank',
    icon: 'build',
  },
  3: {
    key: 3,
    title: 'Code Compiling',
    formattedMessage: 'codeCompiling',
    icon: 'code',
  },
}
