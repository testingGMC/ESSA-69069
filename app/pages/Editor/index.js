/**
 *
 * Editor
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'
import { push } from 'connected-react-router'
import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'
import {
  getInstanceCollaborators,
  getInstantMessages,
  getConnection,
  getConnected,
  getConnectionStatus,
} from 'shared/redux/liveshare/selectors'
import makeSelectEditor, {
  getLearningStyles,
  getAssessments,
  getKeywords,
  getSelections,
} from 'shared/redux/editor/selectors'
import reducer from 'shared/redux/editor/reducer'
import liveShareReducer from 'shared/redux/liveshare/reducer'
import * as actions from 'shared/redux/editor/actions'
import * as liveshareActions from 'shared/redux/liveshare/actions'
import saga from 'shared/redux/editor/saga'
import liveshareSaga from 'shared/redux/liveshare/saga'

import Editor from './Editor'

const EditorIndex = props => {
  useInjectReducer({ key: 'editor', reducer })
  useInjectReducer({ key: 'liveshare', reducer: liveShareReducer })
  return <Editor {...props} />
}

const mapStateToProps = createStructuredSelector({
  skill: makeSelectEditor(),
  learningStyles: getLearningStyles(),
  assessments: getAssessments(),
  keywords: getKeywords(),
  connectedUsers: getInstanceCollaborators(),
  instantMessages: getInstantMessages(),
  connection: getConnection(),
  isConnectedToLiveshare: getConnected(),
  connectionStatus: getConnectionStatus(),
  selections: getSelections(),
})

const mapDispatchToProps = {
  ...actions,
  ...liveshareActions,
  push,
}

const withSaga = injectSaga({ key: 'editor', saga, mode: DAEMON })
const withSagaLiveShare = injectSaga({
  key: 'liveshare',
  saga: liveshareSaga,
  mode: DAEMON,
})
const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
  withSagaLiveShare,
)(EditorIndex)
