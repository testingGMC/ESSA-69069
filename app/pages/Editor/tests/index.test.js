
import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'
import { BrowserRouter } from "react-router-dom";
import EditorIndex from '../index'
import Editor from '../Editor'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'


const initialProps = {
  learningStyles: [],
  clearState: jest.fn(),
  createOrLoadSkillInstance: jest.fn(),
  keywords: [],
  assessments: [],
  connectedUsers: [],
  fetchSkill: jest.fn(),
  updateSkill: jest.fn(),
  addLearningStyle: jest.fn(),
  addPage: jest.fn(),
  deletePage: jest.fn(),
  deleteLearningStyle: jest.fn(),
  updateSelectedLearningStyleId: jest.fn(),
  addAssessment: jest.fn(),
  deleteAssessment: jest.fn(),
  updateSelectedAssessmentId: jest.fn(),
  addKeyword: jest.fn(),
  updateKeyword: jest.fn(),
  deleteKeyword: jest.fn(),
  updateSelectedKeywordId: jest.fn(),
  addReorderChallengeStatement: jest.fn(),
  updateReorderChallengeStatement: jest.fn(),
  deleteReorderChallengeStatement: jest.fn(),
  addQuizQuestion: jest.fn(),
  updateQuizQuestion: jest.fn(),
  deleteQuizQuestion: jest.fn(),
  addQuizChoice: jest.fn(),
  updateQuizChoice: jest.fn(),
  deleteQuizChoice: jest.fn(),
  skill: { data: {} },
  match: {},
  match: {
    params: { skillId: '' }
  }
}

describe('<EditorIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(

      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <EditorIndex {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>
    )
    expect(spy).not.toHaveBeenCalled()
  })


  it('Expect to not log errors in console', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <EditorIndex {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
  })

})

describe('<Editor/>', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Editor {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()

  })

  it('should fetch the editor on mount if the skillId exists', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Editor {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(initialProps.fetchSkill).toHaveBeenCalled()
  })
})