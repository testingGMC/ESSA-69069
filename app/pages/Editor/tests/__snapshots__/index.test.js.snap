// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor/> should render and match the snapshot 1`] = `
<div
  class="editor"
>
  <div />
  <div
    class="ant-row-flex editor-header"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col ant-col-xs-8 ant-col-sm-8 ant-col-md-8 ant-col-lg-8 ant-col-xl-8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="right-actions"
      >
        <i
          aria-label="icon: save"
          class="anticon anticon-save save"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="save"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M893.3 293.3L730.7 130.7c-7.5-7.5-16.7-13-26.7-16V112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V338.5c0-17-6.7-33.2-18.7-45.2zM384 184h256v104H384V184zm456 656H184V184h136v136c0 17.7 14.3 32 32 32h320c17.7 0 32-14.3 32-32V205.8l136 136V840zM512 442c-79.5 0-144 64.5-144 144s64.5 144 144 144 144-64.5 144-144-64.5-144-144-144zm0 224c-44.2 0-80-35.8-80-80s35.8-80 80-80 80 35.8 80 80-35.8 80-80 80z"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-8 ant-col-sm-8 ant-col-md-8 ant-col-lg-8 ant-col-xl-8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="title"
      />
    </div>
    <div
      class="ant-col ant-col-xs-8 ant-col-sm-8 ant-col-md-8 ant-col-lg-8 ant-col-xl-8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="left-actions global-flex-horizontal-end"
      >
        <div
          class="users"
        />
        <div
          class="close global-flex-vertical-horizontal-center"
        >
          <a
            href="/graphnos/skill-network"
          >
            <span>
              Close 
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-spin-nested-loading"
  >
    <div>
      <div
        class="ant-spin ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      class="ant-spin-container ant-spin-blur"
    >
      <div
        class="ant-row-flex editor-content"
        id="editor-content"
        style="margin-left: -12px; margin-right: -12px;"
      >
        <div
          class="ant-col ant-col-xs-24"
          style="padding-left: 12px; padding-right: 12px;"
        >
          <div
            class="text-box-with-arrows"
          >
            <div
              class="text-box"
            >
              <div
                class="add-learning-style"
              >
                <div
                  class="ant-row-flex add-learning-style-row"
                  style="margin-left: -12px; margin-right: -12px;"
                >
                  <div
                    class="ant-col add-learning-style-column ant-col-xs-12"
                    style="padding-left: 12px; padding-right: 12px;"
                  >
                    <div
                      class="learning-style-type"
                    >
                      <i
                        aria-label="icon: file-text"
                        class="anticon anticon-file-text learning-style-type-icon"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="file-text"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M534 352V136H232v752h560V394H576a42 42 0 0 1-42-42zm-22 322c0 4.4-3.6 8-8 8H320c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h184c4.4 0 8 3.6 8 8v48zm200-184v48c0 4.4-3.6 8-8 8H320c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h384c4.4 0 8 3.6 8 8z"
                            fill="#e6f7ff"
                          />
                          <path
                            d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM602 137.8L790.2 326H602V137.8zM792 888H232V136h302v216a42 42 0 0 0 42 42h216v494z"
                            fill="#1890ff"
                          />
                          <path
                            d="M312 490v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H320c-4.4 0-8 3.6-8 8zm192 128H320c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                            fill="#1890ff"
                          />
                        </svg>
                      </i>
                      <div
                        class="learning-style-type-name"
                      >
                        <span>
                          Template with text/image/code
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col add-learning-style-column ant-col-xs-12"
                    style="padding-left: 12px; padding-right: 12px;"
                  >
                    <div
                      class="learning-style-type"
                    >
                      <i
                        aria-label="icon: video-camera"
                        class="anticon anticon-video-camera learning-style-type-icon"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="video-camera"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M136 792h576V232H136v560zm64-488c0-4.4 3.6-8 8-8h112c4.4 0 8 3.6 8 8v48c0 4.4-3.6 8-8 8H208c-4.4 0-8-3.6-8-8v-48z"
                            fill="#e6f7ff"
                          />
                          <path
                            d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226z"
                            fill="#1890ff"
                          />
                          <path
                            d="M208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                            fill="#1890ff"
                          />
                        </svg>
                      </i>
                      <div
                        class="learning-style-type-name"
                      >
                        <span>
                          Template with Video
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="drawer"
        >
          <div
            class="div-primary"
          >
            <i
              aria-label="icon: menu-fold"
              class="anticon anticon-menu-fold"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="menu-fold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
                />
              </svg>
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row-flex editor-footer"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col ant-col-xs-4"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="status disconnected"
      >
        <span>
          Disconnected
        </span>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-16"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="preview"
      >
        <i
          aria-label="icon: code"
          class="anticon anticon-code footer-icon"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="code"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 0 0 308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 0 0-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: desktop"
          class="anticon anticon-desktop footer-icon"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="desktop"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
            />
          </svg>
        </i>
        <i
          aria-label="icon: mobile"
          class="anticon anticon-mobile footer-icon"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="mobile"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M744 62H280c-35.3 0-64 28.7-64 64v768c0 35.3 28.7 64 64 64h464c35.3 0 64-28.7 64-64V126c0-35.3-28.7-64-64-64zm-8 824H288V134h448v752zM472 784a40 40 0 1 0 80 0 40 40 0 1 0-80 0z"
            />
          </svg>
        </i>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-4"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="footer-chat"
      />
    </div>
  </div>
</div>
`;
